### ISSUE TITLE
`bad-test-name` never checks `-`-prefixed throwing tests, so `[] -> it-works` passes while `[] +> it-works` is flagged

### ISSUE THREAD

**Issue by Favixx** [objectionary/lints#1217]
## What happens

`bad-test-name.xsl` selects test objects with an ad-hoc `starts-with(@name, '+')`
instead of the shared `eo:test-name()` function:

https://github.com/objectionary/lints/blob/b004cdbe8ec110b776fbf49755aa9f2a7dfd2f17/src/main/resources/org/eolang/lints/tests/bad-test-name.xsl#L12-L16

```xml
<xsl:for-each select="/object//o[starts-with(@name, '+')]">
  <xsl:variable name="regexp" select="'^(can|cannot|accepts|rejects|stops-on)-'"/>
  <xsl:if test="not(matches(eo:escape-plus(@name), $regexp))">
```

The EO parser marks a truthy test attribute (`+> name`) with a `+` prefix and a
throwing test attribute (`-> name`) with a `-` prefix. `org/eolang/funcs/test-name.xsl`
states that "Both kinds are tests":

https://github.com/objectionary/lints/blob/b004cdbe8ec110b776fbf49755aa9f2a7dfd2f17/src/main/resources/org/eolang/funcs/test-name.xsl#L13-L16

```xml
<xsl:function name="eo:test-name" as="xs:boolean">
  <xsl:param name="name"/>
  <xsl:sequence select="starts-with($name, '+') or starts-with($name, '-')"/>
</xsl:function>
```

The motive of the lint says the rule applies to every unit test object:

> Every unit test object must be named to describe the behavior it verifies, starting
> with one of these prefixes: `can-`, `cannot-`, `accepts-`, `rejects-`, or `stops-on-`.

Because the `for-each` only sees `+`-prefixed names, a throwing test never reaches the
regexp and any name it carries is accepted. `stops-on-` in the allowed prefix list is
the idiomatic name for a throwing test, so the very objects that prefix targets are the
ones the rule skips.

`test-name.xsl` was introduced in f632e25 and used to replace `starts-with(@name, '+')`
in seven lints, including the sibling `misc/incorrect-test-object-name.xsl`.
`bad-test-name.xsl` was merged three hours earlier the same day (ca9cef9) and was not
part of that change, so it still carries the old check.

## Steps to reproduce

`truthy.eo`:

```eo
[] > foo
  [] > bar

  [] +> it-works
```

`throwing.eo`, identical except for the test marker:

```eo
[] > foo
  [] > bar

  [] -> it-works
```

Run `new Source(new EoSyntax(new InputOf(Paths.get(file))).parsed()).defects()` on each
and filter for `bad-test-name`.

The parser produces `name="+it-works"` for the first file and `name="-it-works"` for the
second; the objects are otherwise identical.

## Observed

`truthy.eo`:

```text
[bad-test-name/S WARNING]:4 The name of the test object "+it-works" must start with one of the prefixes "can-,⌴cannot-,⌴accepts-,⌴rejects-,⌴stops-on-"
```

`throwing.eo`: no `bad-test-name` defect at all.

## Expected behavior

`bad-test-name` should apply the prefix rule to throwing tests as well, so
`[] -> it-works` is reported just like `[] +> it-works`.

**Comment by 0crat** [comment]
@favixx Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+149](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).

**Comment by VasilevNStas** [comment]
This one is already fixed on `master` by PR #1244 (`fix(#1243): enforce prefixes per test kind in bad-test-name`, merged 2026-08-11) and shipped in 0.0.61.

`bad-test-name.xsl` now selects test objects via the shared `eo:test-name()` function and applies a prefix rule per test kind:
- positive tests (`+`-prefixed) must start with `can-` or `accepts-`;
- negative/throwing tests (`-`-prefixed) must start with `cannot-`, `rejects-` or `stops-on-`.

I verified the exact cases from this issue against the current lint:

```eo
[] > main
  [] -> it-works   # throwing (negative)
    42 > @
  [] +> it-works   # positive
    42 > @
```

Both produce a `bad-test-name` defect now:

```
[bad-test-name/S] The name of the negative test object "it-works" must start with one of the prefixes "cannot-, rejects-, stops-on-"
[bad-test-name/S] The name of the positive test object "it-works" must start with one of the prefixes "can-, accepts-"
```

Suggest closing.
